ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestrouting.py

```diff
@@ -106,41 +106,62 @@ def add_verify_peers_test(description: str, afi: str, bgp_neighbor_ip: str, safi
-            try:
-                neighbor_peer_group = get_item(
-                    bgp_peer_groups, "name", bgp_neighbor["peer_group"], required=True, var_name=f"name: {bgp_neighbor['peer_group']}"
-                )
-                bgp_neighbor_ip = str(get(bgp_neighbor, "ip_address", required=True))
+            bgp_peer_groups = get(self.hostvars[self.device_name], "router_bgp.peer_groups", [])
+            bgp_neighbors = get(self.hostvars[self.device_name], "router_bgp.neighbors", [])
+
+            for idx, bgp_neighbor in enumerate(bgp_neighbors, start=1):
+                # TODO - this matches legacy eos_validate_state BUT works only for neighbors in peer groups...
+                try:
+                    neighbor_peer_group = get_item(bgp_peer_groups, "name", bgp_neighbor["peer_group"], required=True, var_name=bgp_neighbor["peer_group"])
+                except AristaAvdMissingVariableError as e:
+                    LOGGER.warning("Peer group '%s' dictionary is missing from the 'peer_groups' list of the 'router_bgp' data model.", str(e))
+                    continue
+
+                try:
+                    bgp_neighbor_ip = str(get(bgp_neighbor, "ip_address", required=True))
+                except AristaAvdMissingVariableError as e:
+                    LOGGER.warning("Neighbor entry #%d from the 'neighbors' list of the 'router_bgp' data model is missing the variable '%s'.", idx, str(e))
+                    continue
+
+                try:
+                    neighbor_peer_group_type = get(neighbor_peer_group, "type", required=True)
+                except AristaAvdMissingVariableError as e:
+                    LOGGER.warning(
+                        "Peer group '%s' from the 'peer_groups' list of the 'router_bgp' data model is missing the variable '%s'.",
+                        bgp_neighbor["peer_group"],
+                        str(e),
+                    )
+                    continue
 
-                if get(neighbor_peer_group, "type", required=True) == "ipv4":
+                if neighbor_peer_group_type == "ipv4":
                     add_verify_peers_test(description="ip bgp peer state established (ipv4)", afi="ipv4", safi="unicast", bgp_neighbor_ip=bgp_neighbor_ip)
-                elif get(neighbor_peer_group, "type", required=True) == "evpn":
+                elif neighbor_peer_group_type == "evpn":
                     add_verify_peers_test(description="bgp evpn peer state established (evpn)", afi="evpn", bgp_neighbor_ip=bgp_neighbor_ip)
 
-            except AristaAvdMissingVariableError as e:
-                LOGGER.warning("Variable '%s' is missing. Please validate the Router BGP data model of this host.", str(e))
+        else:
+            LOGGER.warning("Variable 'service_routing_protocols_model' is NOT set to 'multi-agent'. %s is skipped.", self.__class__.__name__)
+            return None
```

Comment:
Could invert if and else and return first

```
          if service_routing_protocols_model != "multi-agent":
            LOGGER.warning("Variable 'service_routing_protocols_model' is NOT set to 'multi-agent'. %s is skipped.", self.__class__.__name__)
            return None
          ...
```

Reply:
Agreed. Fixed in 6fbead698cc97cd00c164b0022debd669188af14